### PR TITLE
Add ddp_cpu to DistributedType Enum

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add `ddp_cpu` to DistributedType Enum

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
I think this is covered by `DistributedType.DDP` and `DeviceType.CPU`

**Comment by kaushikb11** [comment]
> I think this is covered by DistributedType.DDP and DeviceType.CPU

Internally, yes. It is `DDPSpawn` and `cpu`. But for API facing, it is `ddp_cpu` for the Users and I need it to check for `training_type` inputs.

**Comment by awaelchli** [comment]
I can confirm that ddp_cpu implies DDPSpawnPlugin with CPU which is correct even though the name does not reflect that properly. In case this was a point of confusion :)

**Comment by Borda** [comment]
> FYI. This PR is not doing any logical changes. Just having `ddp_cpu` part of the Enum.

yes, but it couples back distributed type and device which we heavily decoupled a few months back... :rabbit:

**Comment by awaelchli** [comment]
@Borda I agree it's unfortunate that "cpu" is included in the distribute type name but it's a legacy thing of the old accelerator naming pattern :)

**Comment by Borda** [comment]
> @Borda I agree it's unfortunate that "cpu" is included in the distribute type name but it's a legacy thing of the old accelerator naming pattern :)

so you say we can't replace the 
```py
distrib == DistributedType.DDP_CPU
```
 by 
```py
distrib == DistributedType.DDP and device == DeviceType.CPU
```
?

**Comment by kaushikb11** [comment]
> yes, but it couples back distributed type and device which we heavily decoupled

Agreed. But as Adrian mentioned, unfortunately, it's legacy now and we need to support it.

But I am planning to add support for `Trainer(training_type="ddp_spawn", accelerator="cpu")` which will be similar to `ddp_cpu`. Then, we could maybe deprecate it. cc @awaelchli

**Review by tchaton — APPROVED** [review]
LGTM !
